tests/gh-actions/test_dex_auth.sh

```
@@ -0,0 +1,137 @@
+#!/bin/bash
+set -e
+
+python3 -m venv /tmp/dex-test-venv
+source /tmp/dex-test-venv/bin/activate
+pip3 install -q requests passlib
+
+if ! kubectl get namespace auth &>/dev/null; then
+  kubectl create namespace auth
+  
+  kustomize build ./common/dex/overlays/oauth2-proxy | kubectl apply -f - || {
+    kustomize build ./common/dex/overlays/oauth2-proxy | grep -v "kind: VirtualService" | kubectl apply -f -
+  }
+  sleep 5
+fi
+
+if ! kubectl get deployment -n auth dex &>/dev/null; then
+  kustomize build ./common/dex/overlays/oauth2-proxy | kubectl apply -f -
+  sleep 5
+fi
+
+if kubectl get pod -l app=dex -n auth 2>/dev/null | grep -q "No resources found"; then
+  kustomize build ./common/dex/overlays/oauth2-proxy | kubectl apply -f -
+  sleep 5
+fi
+
+kubectl wait --for=condition=Ready pod -l app=dex -n auth --timeout=180s
+
+if ! kubectl get secret -n auth dex-secret > /dev/null 2>&1; then
+  kubectl create secret generic dex-secret -n auth --from-literal=DEX_USER_PASSWORD=$(python3 -c 'from passlib.hash import bcrypt; print(bcrypt.using(rounds=12, ident="2y").hash("12341234"))')
+  if kubectl get deployment -n auth dex &>/dev/null; then
+    kubectl rollout restart deployment -n auth dex
+    kubectl wait --for=condition=Available deployment -n auth dex --timeout=180s
+  fi
+fi
```

No, for that there is tests/gh-actions/install_oauth2-proxy.sh and the other scripts you should create as mentioned above. Way too many lines and if else in general. Just do and let the workflow fail if one step fails.

---

KISS principle means Keep it simple and stupid. do not add abstraction where it is not needed.